### PR TITLE
CES-37 update Repeat interval and stop repeating after

### PR DESCRIPTION
**Pull request by VuHoDac**
https://issues.liferay.com/browse/CES-37

### CODE REVIEW

File: modules/apps/powwow/powwow-portlet/docroot/meetings/calendar_booking_recurrence_container.jspf [hunk 1]
```diff
@@ -86,6 +85,7 @@ if (recurrence != null) {
 <div class="calendar-portlet-recurrence-container hide" id="<portlet:namespace />recurrenceContainer">
 	<table>
 		<aui:input name="exceptionDates" type="hidden" value="<%= exceptionDatesString %>" />
+		<aui:input name="intervalValue" type="hidden" value="<%= interval %>" />
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/meetings/calendar_booking_recurrence_container.jspf]: could we rename this field for better understanding? because its name is `intervalValue` and use value of `interval`, while there is another field named interval `<aui:select label="" name="interval">`. that logic is somehow hard to follow

File: modules/apps/powwow/powwow-portlet/docroot/meetings/calendar_booking_recurrence_container.jspf [hunk 2]
```diff
@@ -13,7 +13,6 @@
  * details.
  */
 --%>
-
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/meetings/calendar_booking_recurrence_container.jspf]: basically we should keep this empty line

File: modules/apps/powwow/powwow-portlet/docroot/meetings/calendar_booking_recurrence_container.jspf [hunk 3]
```diff
@@ -195,18 +187,14 @@ if (recurrence != null) {
...
-						<label><liferay-ui:message key="never" /></label>
-					</aui:field-wrapper>
-
 					<aui:field-wrapper cssClass="calendar-portlet-recurrence-limit" label="">
 						<aui:input checked="<%= afterChecked %>" label="<%= StringPool.BLANK %>" name="ends" type="radio" value="after" />
 
 						<label><liferay-ui:message key="after" /></label>
 
-						<aui:input disabled="<%= !afterChecked %>" label="<%= StringPool.BLANK %>"  name="count" size="5" type="text" value="<%= afterChecked ? count : StringPool.BLANK %>" />
+						<aui:input autoFocus="false" disabled="<%= !afterChecked %>" label="<%= StringPool.BLANK %>" name="count" type="number" min="0" max="50" value="<%= afterChecked ? count : StringPool.BLANK %>">
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/meetings/calendar_booking_recurrence_container.jspf]: we can adjust to have min=2, because min=0 or min=1 does not make sense.

File: modules/apps/powwow/powwow-portlet/docroot/meetings/calendar_booking_recurrence_container.jspf [hunk 4]
```diff
@@ -195,18 +187,14 @@ if (recurrence != null) {
...
+							<aui:validator name="max">50</aui:validator>
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/meetings/calendar_booking_recurrence_container.jspf:197]: do we need to add a validator of `min`?

File: modules/apps/powwow/powwow-portlet/docroot/meetings/calendar_booking_recurrence_container.jspf [hunk 5]
```diff
@@ -309,7 +297,11 @@ if (recurrence != null) {
 								label: '<liferay-ui:message key="done" />',
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/meetings/calendar_booking_recurrence_container.jspf:299]: should we think about enable/disable the button `Done` when the violation happens?

File: modules/apps/powwow/powwow-portlet/docroot/meetings/js/recurrence.js [hunk 6]
```diff
@@ -410,6 +413,48 @@
...
+
+							var frequencyValue = instance._getFrequency();
+
+							var interValContainer = instance.get('intervalSelect');
+							interValContainer.all('option').remove();
+
+							var intervalCount = 0;
+
+							if (frequencyValue == FREQUENCY_DAILY) {
+								intervalCount = 90;
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/meetings/js/recurrence.js]: i just found that currently the Powwow meeting support maximum 30 for daily interval. i think we can keep this value of 30. i'll also need to update the requirements in tickets for this.

File: modules/apps/powwow/powwow-portlet/docroot/meetings/js/recurrence.js [hunk 7]
```diff
@@ -410,6 +413,48 @@
 							startDateDatePicker.after('selectionChange', A.bind(instance._onStartDateDatePickerChange, instance));
 						},
 
+						_bindIntervalSelection: function() {
+							var instance = this;
+
+							var frequencyValue = instance._getFrequency();
+
+							var interValContainer = instance.get('intervalSelect');
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/meetings/js/recurrence.js]: why don't we name the variable `intervalSelect`?

File: modules/apps/powwow/powwow-portlet/docroot/meetings/js/recurrence.js [hunk 8]
```diff
@@ -410,6 +413,48 @@
...
+							interValContainer.all('option').remove();
+
+							var intervalCount = 0;
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/meetings/js/recurrence.js]: the name `maxInterval` could make more sense

File: modules/apps/powwow/powwow-portlet/docroot/meetings/js/recurrence.js [hunk 9]
```diff
@@ -410,6 +413,48 @@
 							startDateDatePicker.after('selectionChange', A.bind(instance._onStartDateDatePickerChange, instance));
 						},
 
+						_bindIntervalSelection: function() {
+							var instance = this;
+
+							var frequencyValue = instance._getFrequency();
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/meetings/js/recurrence.js]: var name could be simply `frequency` instead. or maybe `selectedFrequency`

File: modules/apps/powwow/powwow-portlet/docroot/meetings/calendar_booking_recurrence_container.jspf (hunk 3 above)
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/meetings/calendar_booking_recurrence_container.jspf]: beside that, could we see if the error message be highlighted red or yellow?

**Review by hhtank — CHANGES_REQUESTED** [review]

File: modules/apps/powwow/powwow-portlet/docroot/meetings/calendar_booking_recurrence_container.jspf (hunk 5 above)
Comment (VuHoDac) [reply modules/apps/powwow/powwow-portlet/docroot/meetings/calendar_booking_recurrence_container.jspf:299]: It is difficult to enable/disable button of liferay window because we can't identify id of name of button

File: modules/apps/powwow/powwow-portlet/docroot/meetings/css/main.css [hunk 10]
```diff
@@ -1,6 +1,10 @@
 @import "compass";
 @import "mixins";
 
+.help-block {
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/meetings/css/main.css:4]: remove this override as discussed, since the theme already defined its styles.

File: modules/apps/powwow/powwow-portlet/docroot/meetings/js/recurrence.js [hunk 11]
```diff
@@ -410,6 +413,48 @@
...
+
+							var selectedFrequency = instance._getFrequency();
+
+							var intervalSelect = instance.get('intervalSelect');
+							intervalSelect.all('option').remove();
+
+							var maxInterval = 0;
+
+							if (selectedFrequency == FREQUENCY_DAILY) {
+								maxInterval = 15;
```
Comment (hhtank) [inline modules/apps/powwow/powwow-portlet/docroot/meetings/js/recurrence.js:427]: this should be 30

**Review by hhtank — COMMENTED** [review]
merged with minor updates as these 2 last comments from mine.
